migrate/migration.go

```diff
@@ -291,12 +292,22 @@ func WithNopMigration() MigrationOption {
 
 func sortAsc(ms MigrationSlice) {
 	sort.Slice(ms, func(i, j int) bool {
+		ni, ei := strconv.ParseInt(ms[i].Name, 10, 64)
+		nj, ej := strconv.ParseInt(ms[j].Name, 10, 64)
+		if ei == nil && ej == nil && ni != nj {
+			return ni < nj
+		}
```

Comment:
Unfortunately, this does not account for any non-numerical characters in the migration name, e.g. **_init** in `0001_init`, so the error will always be non-nil and the sort will remain lexicographical.

See [Go Playground](https://go.dev/play/p/hDgwmiYHFof)

Reply:
You're missing the fact that the name field only contains the prefix of the migration file name https://github.com/uptrace/bun/blob/04da1b6d1c6371a471d34822bf4d3bc429d57b1e/migrate/migrations.go#L51